File: src/services/applicationService.js
import api from "../api/api";

export const getApplications = () =>
    api.get("/applications");

export const getApplicationById = (id) =>
    api.get(`/applications/${id}`);

export const createApplication = (data) =>
    api.post("/applications", data);

export const updateApplication = (id, data) =>
    api.put(`/applications/${id}`, data);

export const deleteApplication = (id) =>
    api.delete(`/applications/${id}`);

export const submitApplication = (id) =>
    api.post(`/applications/${id}/submit`);

export const approveApplication = (id, remarks) =>
    api.post(`/applications/${id}/approve`, null, {
        params: { remarks }
    });

export const rejectApplication = (id, remarks) =>
    api.post(`/applications/${id}/reject`, null, {
        params: { remarks }
    });

File: backend/src/main/java/com/cdac/application_management_system/service/ApplicationService.java
package com.cdac.application_management_system.service;

import com.cdac.application_management_system.dto.request.ApplicationRequestDTO;
import com.cdac.application_management_system.dto.response.ApplicationResponseDTO;

import java.util.List;

public interface ApplicationService {

    // CREATE
    ApplicationResponseDTO createApplication(ApplicationRequestDTO request);

    // GET BY ID
    ApplicationResponseDTO getApplicationById(Long id);

    // Remaining APIs (we'll convert next)
    List<ApplicationResponseDTO> getAllApplications();

    ApplicationResponseDTO updateApplication(Long id, ApplicationRequestDTO request);

    ApplicationResponseDTO submitApplication(Long id, ApplicationRequestDTO request);

    ApplicationResponseDTO approveApplication(Long id, String remarks);

    ApplicationResponseDTO rejectApplication(Long id, String remarks);

    void deleteApplication(Long id);
}
